frontend/amundsen_application/static/js/ducks/search/filters/sagas.ts

```diff
@@ -14,18 +14,20 @@ import { UpdateSearchFilter, UpdateFilterRequest } from './reducer';
  * Then executes a search on current resource based with new filters and current search state values.
  */
 export function* filterWorker(action: UpdateFilterRequest): SagaIterator {
-  const { categoryId, value } = action.payload;
+  const { searchFilters } = action.payload;
   const state = yield select(getSearchState);
   const { search_term, resource, filters } = state;
   let resourceFilters = {
     ...filters[resource],
   };
 
-  if (value === undefined) {
-    resourceFilters = filterFromObj(resourceFilters, [categoryId]);
-  } else {
-    resourceFilters[categoryId] = value;
-  }
+  searchFilters.forEach((filter) => {
+    if (filter.value === undefined) {
+      resourceFilters = filterFromObj(resourceFilters, [filter.categoryId]);
```

Comment:
Just to make sure, this is not overwriting the previous filters on each of the "searchFilters" items, right?

Reply:
That's correct, each apply will still retain the filters from the previous search (minus any that may have been cleared) and will just add on any new ones in addition